File: java/omnia/data/structure/mutable/MutableDirectedGraph.java
package omnia.data.structure.mutable;

import omnia.data.structure.DirectedGraph;

public interface MutableDirectedGraph<E> extends MutableGraph<E>, DirectedGraph<E> {

  @Override
  DirectedNode<E> addNode(E item);

  @Override
  DirectedNode<E> addNodeIfAbsent(E item);

  void addEdge(E first, E second);

  boolean removeEdge(E first, E second);
}

File: src/omnia/data/structure/mutable/MutableDirectedGraph.kt
package omnia.data.structure.mutable

import omnia.data.structure.DirectedGraph

interface MutableDirectedGraph<E> : MutableGraph<E>, DirectedGraph<E> {
    fun addEdge(from: E, to: E)
    fun removeEdge(from: E, to: E): Boolean
    fun removeEdgeUnknownTyped(from: Any?, to: Any?): Boolean
}